### PR TITLE
Moved parser presets into configuration file #1662

### PR DESCRIPTION
**Pull request by joachimmetz**
Moved parser presets into configuration file #1662

### CODE REVIEW

File: plaso/parsers/presets.py [hunk 1]
```diff
@@ -3,75 +3,105 @@
 
 from __future__ import unicode_literals
 
+import yaml
+
+from plaso.lib import errors
+
+
 class ParserPresetDefinition(object):
```
Comment (Onager) [inline plaso/parsers/presets.py]: Please rename to ParserPreset - the definitions are the values in the Yaml file, the define the preset. Note also line 95 - "GetPresets".

File: plaso/parsers/presets.py [hunk 2]
```diff
@@ -3,75 +3,105 @@
...
+      parsers (list[str]): names of parser and parser plugins.
+    """
     super(ParserPresetDefinition, self).__init__()
-    self.name = None
-    self.parsers = None
+    self.name = name
+    self.parsers = parsers
 
 
 class ParserPresets(object):
```
Comment (Onager) [inline plaso/parsers/presets.py]: Rename to PresetManager or PresetReader or similar. This class doesn't contain preset definitions like it used to.

File: plaso/parsers/presets.py [hunk 3]
```diff
@@ -3,75 +3,105 @@
...
   """Parser and parser plugin presets."""
```
Comment (Onager) [inline plaso/parsers/presets.py]: This docstring also needs to adjust to account for the change in functionality of this class.

File: plaso/parsers/presets.py (hunk 1 above)
Comment (joachimmetz) [reply plaso/parsers/presets.py]: ack

File: plaso/parsers/presets.py (hunk 2 above)
Comment (joachimmetz) [reply plaso/parsers/presets.py]: ack

File: plaso/parsers/presets.py (hunk 3 above)
Comment (joachimmetz) [reply plaso/parsers/presets.py]: ack

File: tests/parsers/presets.py [hunk 4]
```diff
@@ -11,42 +11,78 @@
...
 
 
-class ParserPresetsTest(shared_test_lib.BaseTestCase):
-  """Tests for the parser and parser pluging presets."""
+@shared_test_lib.skipUnlessHasTestFile(['presets.yaml'])
+class ParserPresetsManagerTest(shared_test_lib.BaseTestCase):
+  """Tests for the parser and parser plugin presets manager."""
 
   def testGetNames(self):
     """Tests the GetNames function."""
```
Comment (Onager) [inline tests/parsers/presets.py]: Please add a test that checks that the names are in alphabetical order, per the GetNames docstring.

File: plaso/cli/helpers/filter_file.py [hunk 5]
```diff
@@ -47,6 +47,7 @@ def ParseOptions(cls, options, configuration_object):
 
     Raises:
       BadConfigObject: when the configuration object is of the wrong type.
+      BadConfigOptions: if the collection file does not exist.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/cli/helpers/filter_file.py]: ack

File: plaso/parsers/presets.py [hunk 6]
```diff
@@ -3,75 +3,105 @@
...
+  """Parser and parser plugin preset.
 
   Attributes:
     name (str): name of the preset.
-    parsers (str): names of parser and parser plugins.
+    parsers (list[str]): names of parser and parser plugins.
   """
 
+  def __init__(self, name, parsers):
+    """Initializes a parser and parser plugin preset definition.
```
[the remark this answers is not in the document]
Comment (joachimmetz) [reply plaso/parsers/presets.py]: ack

File: tests/parsers/presets.py (hunk 4 above)
Comment (joachimmetz) [reply tests/parsers/presets.py]: ack

**Pull request merged**